packages/doenetml-worker-rust/lib-doenetml-core/src/core/props/data_query/data_query_type_check.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{
+    components::ComponentVariantPropTypes,
+    props::{prop_profile_to_type, PropProfile},
+};
+
+use super::{DataQuery, PropSpecifier, PropValueType};
+
+impl DataQuery {
+    /// Guess the return type of a data query. Since this return type may depend on
+    /// retrieving a prop from a component, you must pass in a reference to the component.
+    ///
+    /// Note: this function assumes the data query is well formed (E.g., it only uses `PropSpecifier::LocalIdx` in conjunction
+    /// with `PropSource::Me`). If the data query is not well formed, this function may return an incorrect type.
+    #[allow(clippy::result_unit_err)]
+    pub fn guess_return_type<T: ComponentVariantPropTypes + 'static>(
+        &self,
+    ) -> Result<Vec<PropValueType>, ()> {
```

Comment:
For the queries that include matching `PropProfile`s, the result is a vector of props, each of which could be one of the `PropType`s from one of the `PropProfile`s. It is not a list of props which match the `PropProfile`s in sequence.

Reply:
I don't understand this comment...`Vec<PropValueType>` is meant to store the union type of several `PropValueType` variants.

Reply:
Nice. That's good to know. I thought the "return type of a data query" was a vector saying the types of each component of the returned vector; I didn't realize it was a union. I haven't gotten to the implementation part of the testing feature yet, as you can tell.